docs/contributing/coding-guidelines/ros-nodes/directory-structure.md

```diff
@@ -1,75 +1,242 @@
+autoware_gnss_poser
+├─ package.xml
+├─ CMakeLists.txt
+├─ README.md
+│
 ├─ config
-│   ├─ foo_ros.param.yaml
-│   └─ foo_non_ros.yaml
+│   ├─ gnss_poser.param.yaml
+│   └─ another_non_ros_config.yaml
+│
+├─ schema
+│   └─ gnss_poser.schema.json
+│
 ├─ doc
 │   ├─ foo_document.md
 │   └─ foo_diagram.svg
-├─ include
-│   └─ <package_name>
-│       └─ foo_public.hpp
-├─ launch
-│   ├─ foo.launch.xml
-│   └─ foo.launch.py
-├─ schema
-│   └─ foo_node.schema.json
+│
+├─ include  # for exporting headers
+│   └─ autoware
+│       └─ gnss_poser
+│           └─ exported_header.hpp
+│
 ├─ src
-│   ├─ foo_node.cpp
-│   ├─ foo_node.hpp
-│   └─ foo_private.hpp
-├─ test
-│   └─ test_foo.cpp
+│   ├─ include
+│   │   ├─ gnss_poser_node.hpp
+│   │   └─ foo.hpp
+│   ├─ gnss_poser_node.cpp
+│   └─ bar.cpp
+│
+├─ launch
+│   ├─ gnss_poser.launch.xml
+│   └─ gnss_poser.launch.py
+│
+└─ test
+    ├─ test_foo.hpp  # or place under an `include` folder here
+    └─ test_foo.cpp
+```
+
+### Package name
+
+- All the packages in Autoware should be prefixed with `autoware_`.
+- Even if the package is exports a node, the package name **should NOT** have the `_node` suffix.
+- The package name should be in `snake_case`.
+
+| Package Name                      | OK  | Alternative                  |
+| --------------------------------- | --- | ---------------------------- |
+| path_smoother                     | ❌  | autoware_path_smoother       |
+| autoware_trajectory_follower_node | ❌  | autoware_trajectory_follower |
+| autoware_geography_utils          | ✅  | -                            |
+
+### Package folder
+
+```txt
+autoware_gnss_poser
 ├─ package.xml
 ├─ CMakeLists.txt
 └─ README.md
 ```
 
-### Directory descriptions
+The package folder name should be the same as the package name.
 
-#### `config`
+#### `package.xml`
+
+- The package name should be entered within the `<name>` tag.
+  - `<name>autoware_gnss_poser</name>`
 
-Place configuration files such as node parameters.
-For ROS parameters, use the extension `.param.yaml`.
-For non-ROS parameters, use the extension `.yaml`.
+#### `CMakeLists.txt`
 
-Rationale: Since ROS parameters files are type-sensitive, they should not be the target of some code formatters and linters. In order to distinguish the file type, we use different file extensions.
+- The [`project()`](https://cmake.org/cmake/help/latest/command/project.html) command should call the package name.
+  - **Example:** `project(autoware_gnss_poser)`
+
+##### Exporting a composable node component executable
+
+```cmake
+ament_auto_add_library(${PROJECT_NAME} SHARED
+  src/gnss_poser_node.cpp
+)
+
+rclcpp_components_register_node(${PROJECT_NAME}
+  PLUGIN "autoware::gnss_poser::GNSSPoserNode"
+  EXECUTABLE ${PROJECT_NAME}_node
+)
+```
 
-#### `doc`
+- The composable node component executable:
+  - should have `_node` suffix.
+  - should start with `${PROJECT_NAME}`
 
-Place document files and link from README.
+##### Exporting a standalone node executable
```

Comment:
I don't think we should allow non-component executables, or if we do, discourage them strongly and require the author a justification. We've seen many advantages in component executables vs non-component, such as predictability, code uniformity, standard tooling, etc.

Reply:
Added information for discouraging and limiting the use of non-composable nodes with: https://github.com/autowarefoundation/autoware-documentation/commit/9ba71b95d580d47e711653315fd43c8616161241